src/components/craft/settings/ImageSettings.tsx

```diff
@@ -33,18 +104,52 @@ const ImageSettings: CraftElementSettings<ImageProps> = ({ props, onPropsChange
     <Form form={form} layout="vertical" colon={false} requiredMark={false} onChange={handleChange}>
       <SizeStyleInput
         value={props.customStyle}
+        imgProps={{
+          width: imgWidth,
+          height: imgHeight,
+          aspectRatio: aspectRatio,
+          gcd: imgGcd,
+          widthAspect: widthAspect,
+          heightAspect: heightAspect,
+        }}
+        isImgAutoHeight={isImgAutoHeight}
         onChange={value =>
           onPropsChange?.({ ...props, customStyle: { ...props.customStyle, ...value }, ratio: undefined })
         }
       />
+      <Form.Item>
+        <Checkbox
+          checked={isImgAutoHeight}
+          onChange={e => {
+            setIsImgAutoHeight(!isImgAutoHeight)
+            if (typeof props.customStyle?.width === 'string') {
+              let width =
```

Comment:
If a variable is immutable use `const` to declare directly is more clear.

Reply:
had modified in below commit.
thanks.